docs/10_Overview_And_Usage.md

```diff
@@ -0,0 +1,176 @@
+# ai-rag-template
+
+The `ai-rag-template` is a template to build and run your own RAG application to have a Chatbot that is capable to perform a conversation with an user.
+
+The service is developed using the [LangChain](https://python.langchain.com/docs/get_started/introduction/) framework, which enables creating sequences of complex interactions using Large Language Models. The web server was implemented using the [FastAPI](https://fastapi.tiangolo.com/) framework.
+
+In order to work, it is required to have a MongoDB instance to be used as Vector Store and that supports [MongoDB Vector Search indexes](https://www.mongodb.com/docs/atlas/atlas-vector-search/tutorials/vector-search-quick-start/), which means an Atlas instance with version 6 or above.
+
+## Overview
+
+The following is the high-level architecture of _ai-rag-template_.
+
+```mermaid
+flowchart LR
+  fe[Frontend]
+  be[Backend]
+  vs[(Vector Store)]
+  llm[LLM API]
+  eg[Embeddings Generator API]
+
+  fe --1. user question +\nchat history--> be
+  be --2. user question--> eg
+  eg --3. embedding-->be
+  be --4. similarity search-->vs
+  vs --5. similar docs-->be
+  be --6. user question +\nchat history +\nsimilar docs-->llm
+  llm --7. bot answer--> be
+  be --8. bot answer--> fe
+```
+
+### Embeddings
+
+Please mind that the _ai-rag-template_ does not include embeddings or any logic to create them. It is intended that the Vector Store will include the embeddings (or these are generated separately). In any case, please ensure that the embedding model used the populate the Vector Store is the same embedding model used when running the service, otherwise the service will generate answers only based on its own knowledge, without being able to use the Vector Store, with the risk of hallucinations when chatting with the user.
```

Comment:
```suggestion
Please mind that the template does not include embeddings or any logic to create them. It is intended that the Vector Store will include the embeddings (or these are generated separately). In any case, please ensure that the embedding model used the populate the Vector Store is the same embedding model used when running the service, otherwise the service will generate answers only based on its own knowledge, without being able to use the Vector Store, with the risk of hallucinations when chatting with the user.
```